src/test/java/com/iota/iri/service/TipsManagerTest.java

```diff
@@ -65,111 +75,269 @@ public static void tearDown() throws Exception {
+        });
     }
 
     @Test
-    public void updateRatingsSerialWorks2() throws Exception {
-        Hash[] hashes = new Hash[5];
-        hashes[0] = getRandomTransactionHash();
-        new TransactionViewModel(getRandomTransactionTrits(), hashes[0]).store(tangle);
-        for(int i = 1; i < hashes.length; i ++) {
-            hashes[i] = getRandomTransactionHash();
-            new TransactionViewModel(getRandomTransactionWithTrunkAndBranch(hashes[i-1], hashes[i-(i > 1 ?2:1)]), hashes[i]).store(tangle);
-        }
-        Map<Hash, Long> ratings = new HashMap<>();
-        tipsManager.recursiveUpdateRatings(hashes[0], ratings, new HashSet<>());
-        Assert.assertTrue(ratings.get(hashes[0]).equals(12L));
+    public void testCalculateCommulativeWeightWithLeftBehind() throws Exception {
+        TransactionViewModel transaction, transaction1, transaction2, transaction3, transaction4;
+        transaction = new TransactionViewModel(getRandomTransactionTrits(), getRandomTransactionHash());
+        transaction1 = new TransactionViewModel(getRandomTransactionWithTrunkAndBranch(transaction.getHash(),
+                transaction.getHash()), getRandomTransactionHash());
+        transaction2 = new TransactionViewModel(getRandomTransactionWithTrunkAndBranch(transaction1.getHash(),
+                transaction.getHash()), getRandomTransactionHash());
+        transaction3 = new TransactionViewModel(getRandomTransactionWithTrunkAndBranch(transaction2.getHash(),
+                transaction.getHash()), getRandomTransactionHash());
+        transaction4 = new TransactionViewModel(getRandomTransactionWithTrunkAndBranch(transaction3.getHash(),
+                transaction1.getHash()), getRandomTransactionHash());
+        Set<Hash> approvedHashes = new HashSet<>();
+        transaction.store(tangle);
+        transaction1.store(tangle);
+        approvedHashes.add(transaction2.getHash());
+        transaction2.store(tangle);
+        approvedHashes.add(transaction3.getHash());
+        transaction3.store(tangle);
+        transaction4.store(tangle);
+
+        Map<Buffer, Integer> cumulativeWeight = tipsManager.calculateCumulativeWeight(approvedHashes,
+                transaction.getHash(), true, new HashSet<>());
+
+        log.info(cumulativeWeight.toString());
+        String msg = "Cumulative weight is wrong for tx";
+        Assert.assertEquals(msg + 4, 1, cumulativeWeight.get(transaction4.getHash().getSubHash()).intValue());
+        Assert.assertEquals(msg + 3, 1, cumulativeWeight.get(transaction3.getHash().getSubHash()).intValue());
+        Assert.assertEquals(msg + 2, 1, cumulativeWeight.get(transaction2.getHash().getSubHash()).intValue());
+        Assert.assertEquals(msg + 1, 2, cumulativeWeight.get(transaction1.getHash().getSubHash()).intValue());
+        Assert.assertEquals(msg + 0, 3, cumulativeWeight.get(transaction.getHash().getSubHash()).intValue());
     }
 
-    //@Test
+    //    @Test
```

Comment:
I recommend removing changes to this function, if it's only formatting and comments